src/lib/libembind.js

```diff
@@ -558,10 +559,10 @@ var LibraryEmbind = {
               }
               HEAPU8[ptr + i] = charCode;
             }
-          } else {
-            for (var i = 0; i < length; ++i) {
-              HEAPU8[ptr + i] = value[i];
-            }
+          }
+        } else {
+          for (var i = 0; i < length; ++i) {
+            HEAPU8[ptr + i] = value[i];
           }
```

Comment:
This can (probably) be further simplified to:
```suggestion
          HEAPU8.set(value, ptr);
```

Reply:
Nice!